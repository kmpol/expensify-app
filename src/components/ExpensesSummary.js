import React from 'react'
import { connect } from 'react-redux';
import selectedExpenses from '../selectors/expenses'
import selectedExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component {

    render() {
        const grammarExpenseDisplay = this.props.expenseCount === 1 ? "expense" : "expenses"
        return (
            <div>
                <p>Viewing {this.props.expenseCount} {grammarExpenseDisplay} with total: ${numeral(this.props.expensesInTotal / 100).format('0.00')}</p>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectedExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expensesInTotal: selectedExpensesTotal(visibleExpenses)
    }
}
export default connect(mapStateToProps)(ExpensesSummary)