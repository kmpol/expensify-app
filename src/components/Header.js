import { NavLink } from 'react-router-dom'
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux'

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink activeClassName="is-active" to="/dashboard">Home page</NavLink>
        <NavLink activeClassName="is-active" to="/create">Create expense</NavLink>
        <NavLink activeClassName="is-active" to="/help">Help page</NavLink>
        <button onClick={startLogout()}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout)
    }
}

export default connect(undefined, mapDispatchToProps)(Header)