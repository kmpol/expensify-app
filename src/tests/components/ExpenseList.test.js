import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json'

import { ExpenseList } from '../../components/ExpenseList'
import expenses from '../fixtures/expenses'

configure({ adapter: new Adapter() });
test('Should render expenses list with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})

test('Should render expenses list with empty message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
})