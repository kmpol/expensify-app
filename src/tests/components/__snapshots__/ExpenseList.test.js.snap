// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expenses list with empty message 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <div>
    <h2>
      No expneses
    </h2>
  </div>
</div>
`;

exports[`Should render expenses list with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <div>
    <ExpenseListItem
      amount={123}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      note="1"
    />
    <ExpenseListItem
      amount={123123}
      createdAt={-345600000}
      description="Rent"
      id="2"
      key="2"
      note="2"
    />
    <ExpenseListItem
      amount={9999}
      createdAt={345600000}
      description="Credit card"
      id="3"
      key="3"
      note="3"
    />
  </div>
</div>
`;
